### PR TITLE
[Proof-of-Concept] Get rid of bind* functions

### PR DESCRIPTION
**Pull request by yumauri**
This PR is just to show the differences, you can take an idea :)

### CODE REVIEW

File: src/effector/createUnit.js [hunk 1]
```diff
@@ -279,7 +278,7 @@ export function createStore<State>(
       } else {
         eventOrFn(store.getState())
       }
-      return watchUnit(store, eventOrFn)
+      return watchUnit(store)(eventOrFn)
```
Comment (zerobias) [inline src/effector/createUnit.js]: No `fn()()`, please

Comment (yumauri) [reply src/effector/createUnit.js]: What about like this? :)
```suggestion
      const watchStore = watchUnit(store)
      return watchStore(eventOrFn)
```

Comment (zerobias) [reply src/effector/createUnit.js]: Nice try 😄But no, no `fn()()`, please

Comment (yumauri) [reply src/effector/createUnit.js:281]: Hey, you have `fn()()` inside `createEffect` ;) And that wasn't my work :)

image

**Comment by yumauri** [comment]
Reckon, idea was taken and considered, so I'll close this PR to clear some space and reduce amount of chaos in the universe :)

**Pull request closed without merging**